### PR TITLE
[11.0][base_exception] refactor tests

### PR DESCRIPTION
**Pull request by JordiBForgeFlow**
Fixes https://github.com/OCA/server-tools/issues/1110

### CODE REVIEW

**Comment by JordiBForgeFlow** [comment]
@mourad-ehm can you review?

File: base_exception/tests/test_base_exception.py [hunk 1]
```diff
@@ -2,43 +2,116 @@
...
 
-        self.base_exception = self.env['base.exception']
-        self.exception_rule = self.env['exception.rule']
-        self.exception_confirm = self.env['exception.rule.confirm']
+    @classmethod
+    def setUpClass(cls):
+        super(TestBaseException, cls).setUpClass()
 
-        self.exception_rule._fields['rule_group'].selection.append(
+        class PurchaseTest(models.Model):
```
Comment (simahawk) [inline base_exception/tests/test_base_exception.py]: I'd move this to a separated file anyway and here run only `init_test_models`

File: base_exception/tests/test_base_exception.py [hunk 2]
```diff
@@ -47,18 +120,18 @@ def setUp(self):
...
             'name': "Qty > 0",
             'sequence': 10,
             'rule_group': "test_base",
             'model': "base.exception.test.purchase.line",
             'code': """if test_base_line.qty <= 0:
         failed=True"""})
 
-    def test_sale_order_exception(self):
-        partner = self.env.ref('base.res_partner_1')
+    def test_sale_order_exception(cls):
```
Comment (simahawk) [inline base_exception/tests/test_base_exception.py]: why is this wrapped into setUpClass?

File: base_exception/tests/test_base_exception.py [hunk 3]
```diff
@@ -2,43 +2,116 @@
...
 from odoo.tests import common
+from odoo import api, fields, models
+from .common import init_test_models
 
 import logging
 
 _logger = logging.getLogger(__name__)
 
 
-# @common.at_install(False)
```
Comment (simahawk) [inline base_exception/tests/test_base_exception.py:11]: so this trick plays nice only if you run the whole test suite at install I guess?

**Review by simahawk — CHANGES_REQUESTED** [review]

File: base_exception/tests/test_base_exception.py (hunk 3 above)
Comment (JordiBForgeFlow) [reply base_exception/tests/test_base_exception.py:11]: This is the way I've seen it being done in https://github.com/OCA/mis-builder/blob/10.0/mis_builder/tests/test_kpi_data.py#L19, and seems to be no issue testing during install, which is the default method and easiest to verify.

**Comment by JordiBForgeFlow** [comment]
@simahawk Many thanks for your review. I have attended to your comments.

File: base_exception/tests/test_base_exception.py (hunk 2 above)
Comment (JordiBForgeFlow) [reply base_exception/tests/test_base_exception.py]: No need. I fixed

File: base_exception/tests/test_base_exception.py (hunk 1 above)
Comment (JordiBForgeFlow) [reply base_exception/tests/test_base_exception.py]: Done.

File: base_exception/tests/test_base_exception.py (hunk 3 above)
Comment (simahawk) [reply base_exception/tests/test_base_exception.py:11]: hmm then I guess that if for any reason we'll need to run a test after installation we could use a different class and make sure this one is loaded before

**Review by simahawk — APPROVED** [review]
@jbeficent can you squash?

**Comment by simahawk** [comment]
@mourad-ehm ping :)

**Review by mourad-ehm — APPROVED** [review]

**Comment by mourad-ehm** [comment]
Hi @jbeficent,  Thanks for this PR

**Comment by JordiBForgeFlow** [comment]
@mourad-ehm @simahawk I have now cleaned up in e3658cc715b132643af2b65c6286d3c4f6178a48 the way to create and tear down test classes, following the proposal made by @sbidoul in https://github.com/OCA/mis-builder/pull/67

**Comment by mourad-ehm** [comment]
OK :+1:

**Comment by JordiBForgeFlow** [comment]
@simahawk squashed commits. Will you merge now?

**Comment by simahawk** [comment]
@jbeficent I wish I could do it :)
Nevertheless, pls, make the commit more explicit by including the module name (as you did for the PR title), tnx :)

**Pull request merged**